Sail/src/Sail/entities/systems/Graphics/CrosshairSystem.cpp

```diff
@@ -0,0 +1,55 @@
+#include "pch.h"
+#include "CrosshairSystem.h"
+#include "../../ECS.h"
+#include "Sail/entities/components/Components.h"
+
+
+CrosshairSystem::CrosshairSystem() {
+	registerComponent<CrosshairComponent>(true, true, true);
+
+	m_settings = &Application::getInstance()->getSettings();
+}
+
+CrosshairSystem::~CrosshairSystem() {
+
+}
+
+void CrosshairSystem::update(float dt) {
+	for (auto& e : entities) {
+		CrosshairComponent* c = e->getComponent<CrosshairComponent>();
+
+		// get settings from settingStorage
+		applySettings(c);
+
+		// Update the crosshair if 'i' hit someone
+		if (c->currentlyAltered) {
+			alterCrosshair(e, dt);
+		}
+	}
+}
+
+void CrosshairSystem::applySettings(CrosshairComponent* c) {
+	auto& dynamic = m_settings->applicationSettingsDynamic;
+
+	c->thickness = dynamic["Crosshair"]["Thickness"].value;
+	c->centerPadding = dynamic["Crosshair"]["CenterPadding"].value;
+	c->size = dynamic["Crosshair"]["Size"].value;
+	c->color.x = dynamic["Crosshair"]["Color R"].value;
+	c->color.y = dynamic["Crosshair"]["Color G"].value;
+	c->color.z = dynamic["Crosshair"]["Color B"].value;
+	c->color.w = dynamic["Crosshair"]["Color A"].value;
```

Comment:
Probably no big performance hit, but
    auto& crosshairSetting = dynamic["Crosshair"];
could be added to lower unnecessary lookups in the map.

Reply:
Fixed